packages/subscription-manager/src/components/index.d.ts

```diff
@@ -113,3 +114,20 @@ declare module 'calypso/components/forms/form-label' {
 declare module 'calypso/components/localized-moment' {
 	export const useLocalizedMoment: () => import('moment');
 }
+
+declare module 'calypso/components/localized-moment/provider' {
+	const MomentProvider: FC< { currentLocale: string } >;
+	export default MomentProvider;
+}
+
+declare module 'calypso/state' {
```

Comment:
From my understanding, it doesn't make much sense for a package to depend on the client. It should be the other way around. Therefore, it shouldn't make sense for this package to declare modules that live in the `calypso` client.

Furthermore, declaring them here could create problems and/or conflicts with other packages or the client one that also declare them or rely on them.